src/Workspaces/Core/Portable/Diagnostics/DiagnosticDataLocation.cs

```diff
@@ -13,15 +14,22 @@ namespace Microsoft.CodeAnalysis.Diagnostics
     [DataContract]
     internal sealed class DiagnosticDataLocation
     {
+        /// <summary>
+        /// Path to where the diagnostic was originally reported.  May be a path to a document in a project, or the
+        /// project file itself.
+        /// </summary>
         [DataMember(Order = 0)]
-        public readonly DocumentId? DocumentId;
+        public readonly string OriginalFilePath;
```

Comment:
Can you clarify whether a path is provided by Roslyn itself for diagnostics that don't specify a path, or if the caller (user code) is expected to update their implementations to include a path instead of leaving it blank? What happens for analyzers that previously reported project diagnostics without a location?